### PR TITLE
[WIP] feature/code-coverage adding karma runner

### PR DESCRIPTION
**Pull request by ernestomancebo**
**[Work in progress]**
The intention of this PR is to measure the code covered with unit testing

Pending:
* There's a test failing when running with karma, but passes with Jasmine
* The coverage report maps the .ts file with the generated .js file

### CODE REVIEW

**Comment by btroncone** [comment]
I'm keeping an eye on this, thanks! 👍

File: package.json [hunk 1]
```diff
@@ -8,10 +8,13 @@
     "prepublish": "npm run build_dist",
     "clean:test": "rimraf tmp",
     "clean:dist": "rimraf dist",
+    "clean:coverage": "rimraf coverage",
+    "clean:all": "npm run clean:test && npm run clean:dist&& npm run  clean:coverage",
```
Comment (PhilippMeissner) [inline package.json]:
```suggestion
    "clean:all": "npm run clean:test && npm run clean:dist && npm run clean:coverage",
```

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                pattern: 'src/**/*.ts'
+            },
+            {
+                pattern: 'spec/**/*.ts'
+            }
+        ],
+        plugins: [
+            'karma-jasmine',
+            'karma-typescript',
+            'karma-coverage',
```
Comment (PhilippMeissner) [inline karma.conf.js]: Isn't this quite the duplicate? `karma-coverage` & `karma-coverage-istanbul-reporter`

**Comment by BBlackwo** [comment]
Thanks for the PR @ernestomancebo. 

It seems like this PR was never fully completed, so I'm going to close it for now.

Also, #147 adds coverage with `nyc` so we probably don't need to add karma anymore.

**Pull request closed without merging**